woocommerce/woocommerce-gutenberg-products-block
# Fix use of sanitizeHTML

This PR fixes the `sanitizeHTML` return value data type and imports to the `store-notices-container` so we don't have two same methods in one repo.

Fixes #7230 


### Screenshots



| Before | After |
| ------ | ----- |
|<img width="1006" alt="Screen Shot 2022-09-26 at 12 17 39 PM" src="https://user-images.githubusercontent.com/56378160/192390815-3672e229-1491-4eed-bf71-2deaa3c7e0f8.png">|<img width="763" alt="Screen Shot 2022-09-26 at 12 34 36 PM" src="https://user-images.githubusercontent.com/56378160/192390876-271535a3-502b-4bca-b858-0ca8b56bc641.png">|

### Testing
1. Check out branch 1287-gh-Automattic/woopay
1. Update the code:
In src/Checkout/PaymentsHandler.php, comment or remove the if around the exception:
```
 // if ( empty( $customer_id ) ) {
          $session_data      = WooPaySession::UNSANITIZED_get_data();
          $blog_checkout_url = esc_url( add_query_arg( 'skip_platform_checkout', 'true', $session_data['store_data'] 
                                                 ['blog_checkout_url'] ) );
          /* translators: %1$s opening anchor tag with url, %2$s closing anchor tag */
          $error_message = __( 'we were unable to process your payment. Please %1$s go back to the merchant store %2$s and 
                                          log into WooPay again.', 'woopay' );
          $open_tag      = '<a href="' . $blog_checkout_url . '">';
          $close_tag     = '</a>';
          throw new InvalidSessionException(
	        sprintf( $error_message, $open_tag, $close_tag ),
          );
// }
```
1. Get a valid checkout_session by checking out with a valid email and filling in the OTP code.
1. Place an order and see the error message

### WooCommerce Visibility




### Changelog
> Ensure error messages containing HTML are shown correctly in the Cart and Checkout blocks
